### PR TITLE
Added react router, also changed the name of a class in postsPage

### PR DESCRIPTION
**Pull request by y0w1b**
- [x] All features are tested
- [x] Version is bumped
- [x] Changelog is updated
- [x] Documentation is updated if needed

### CODE REVIEW

File: src/AppRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import PostPage from "./frontend/posts_page/components/App";
+import ChatRoom from "./frontend/chat_room/components/App";
+import Settings from "./frontend/settings/components/App";
+
+class AppRouter extends Component {
+    render() {
+        return (
+            <Router>
+                <div>
+                    <Switch>
```
Comment (ZijiaZhang) [inline src/AppRouter.tsx:18]: I think it is better to add switch to the main `App.ts`. That way we don't need to include the sidebars in all these files. And the code will be cleaner.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 All changes will be logged here
 
+## 0.4.1 [2020-06-08]
```
Comment (ZijiaZhang) [inline CHANGELOG.md:5]: should be `0.5.0`. See [Semantic Versioning](https://semver.org/) Also need version bump in `package.json`.

File: src/AppRouter.tsx (hunk 1 above)
Comment (ZijiaZhang) [reply src/AppRouter.tsx:18]: Like 
``` typescript
render() {
    return (
        <left_panel\>
        <switch>
            <...routes ...>
        </switch>
        <right_panel/>);
}
```

Comment (y0w1b) [reply src/AppRouter.tsx:18]: If I understand it correctly, left_panel and right_panel stuffs refer to the shared side bars and headers right? In that case, the shared components are guaranteed to display on every page.

File: CHANGELOG.md (hunk 2 above)
Comment (y0w1b) [reply CHANGELOG.md:5]: Got it!

**Pull request merged**